docs/aws/audit/esmonitoring/rules/elasticache_repl_grp_auto_failover_enabled_remediation.mdx

```diff
@@ -1,6 +1,203 @@
+   ```
+
+4. **Run the Script**:
+   Execute the script to ensure that Automatic Failover is enabled for all your Elasticache replication groups.
+
+   ```bash
+   python enable_automatic_failover.py
+   ```
+
+This script will iterate through all your Elasticache replication groups, check if Automatic Failover is enabled, and enable it if it is not. This ensures that your Elasticache clusters are configured correctly to handle failovers automatically.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the ElastiCache dashboard.
+
+2. In the navigation pane, choose "Redis" to access the Redis clusters page.
+
+3. On the Redis clusters page, select the Redis cluster that you want to examine.
+
+4. In the lower panel, under the "Details" tab, check the "Automatic Failover" configuration attribute. If the attribute value is set to "disabled", the selected Redis cluster does not have automatic failover enabled.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start, you need to install the AWS CLI on your local machine. You can download it from the official AWS website. After installation, you need to configure it with your AWS account using the command `aws configure`. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all Elasticache clusters: Use the following command to list all the Elasticache clusters in your AWS account.
+
+   ```
+   aws elasticache describe-cache-clusters --query 'CacheClusters[*].[CacheClusterId]'
+   ```
+
+   This command will return a list of all the CacheClusterIds in your AWS account.
+
+3. Check the Automatic Failover status: For each CacheClusterId, you need to check the Automatic Failover status. Use the following command to do this:
+
+   ```
+   aws elasticache describe-replication-groups --query 'ReplicationGroups[*].[ReplicationGroupId, AutomaticFailover]'
+   ```
+
+   This command will return a list of all the ReplicationGroupIds and their Automatic Failover status.
+
+4. Analyze the output: If the Automatic Failover status is 'enabled' for all the ReplicationGroupIds, then there is no misconfiguration. If any ReplicationGroupId has the Automatic Failover status as 'disabled', then it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if Automatic Failover is enabled for AWS ElastiCache, you can use the AWS SDK for Python (Boto3). Here are the steps:
+
+1. **Import the necessary modules and create a session:**
+
+    You need to import Boto3 and create a session using your AWS credentials.
+
+    ```python
+    import boto3
+
+    session = boto3.Session(
+        aws_access_key_id='YOUR_ACCESS_KEY',
+        aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>